signer/core/apitypes/types.go

```diff
@@ -501,7 +501,15 @@ func (typedData *TypedData) encodeArrayValue(encValue interface{}, encType strin
 	for _, item := range arrayValue {
 		if reflect.TypeOf(item).Kind() == reflect.Slice ||
 			reflect.TypeOf(item).Kind() == reflect.Array {
-			encodedData, err := typedData.encodeArrayValue(item, parsedType, depth+1)
+			var (
+				encodedData hexutil.Bytes
+				err         error
+			)
+			if reflect.TypeOf(item).Elem().Kind() == reflect.Uint8 {
+				encodedData, err = typedData.EncodePrimitiveValue(parsedType, item, depth+1)
+			} else {
+				encodedData, err = typedData.encodeArrayValue(item, parsedType, depth+1)
+			}
```

Comment:
```suggestion
			var (
				encodedData hexutil.Bytes
				err         error
			)
			if reflect.TypeOf(item).Elem().Kind() == reflect.Uint8 {
				// the item type is bytes.  encode the bytes array directly instead of recursing. 
				encodedData, err = typedData.EncodePrimitiveValue(parsedType, item, depth+1)
			} else {
				encodedData, err = typedData.encodeArrayValue(item, parsedType, depth+1)
			}
```